Add support for cross bucket replication feature (#12562)
GoogleCloudPlatform/magic-modules: mmv1/third_party/terraform/website/docs/r/storage_transfer_job.html.markdown

* `description` - (Required) Unique description to identify the Transfer Job.

> * `transfer_spec` - (Optional) Transfer specification. Structure [documented below](#nested_transfer_spec). Only one of `transfer_spec`, or `replication_spec` can be specified.

Reviewer: ```suggestion
* `transfer_spec` - (Optional) Transfer specification. Structure [documented below](#nested_transfer_spec). One of `transfer_spec`, or `replication_spec` must be specified.
```

Author: Done

> * `replication_spec` - (Optional) Replication specification. Structure [documented below](#nested_replication_spec). User should not configure
> `transfer_spec`, `schedule`, `event_stream` with this argument.

Reviewer: ```suggestion
* `replication_spec` - (Optional) Replication specification. Structure [documented below](#nested_replication_spec). User should not configure
`transfer_spec`, `schedule`, `event_stream` with this argument. One of `transfer_spec`, or `replication_spec` must be specified.
```

Author: Done
